### PR TITLE
fix: Fix linting of global symbols

### PR DESCRIPTION
**Pull request by hauserx**
Previous implementation was passing empty prelude list to lint.
The starlark-rust does not create warnings for undefined global
symbols in such case.

In general, for bazel we do not have prelude, just a list of global
symbols. Needed to add manually some missing symbols, because bazel does
not report all global symbols in `bazel info build-language`, see
https://github.com/bazel-contrib/vscode-bazel/issues/1#issuecomment-2036369868

In the process:
 - Removed unused prelude. Looks it was carried over with migrating from
   starlark-rust, but prelude is not used in case of bazel.
 - Removed ContextMode run - bazel lsp is not able to actually run any
   starlark, those seems to be leftovers from starlark-rust

### CODE REVIEW

**Comment by cameron-martin** [comment]
Is this ready to review?

**Comment by hauserx** [comment]
I'll get back to it tomorrow, want to check whether caching can be simply implemented (tested on few repos and seems it's not really noticable, but still nice to have feature).

**Comment by cameron-martin** [comment]
I would be for caching being a separate change, since I'm sure some thought is needed around how to invalidate correctly.

**Comment by hauserx** [comment]
Should be ready to review then, please take a look at it, thanks!

File: src/bazel.rs [hunk 1]
```diff
@@ -1203,9 +1129,9 @@ mod tests {
             module.members.iter().any(|(member, _)| member == value)
         }
 
-        // let module = context.get_environment(&LspUrl::File(PathBuf::from("/foo/bar.bzl")));
+        let module = context.get_environment(&LspUrl::File(PathBuf::from("/foo/bar.bzl")));
 
-        // assert!(module_contains(&module, "cc_library"));
+        assert!(module_contains(&module, "cc_library"));
```
Comment (cameron-martin) [inline src/bazel.rs:1134]: Won't this mean that bzl files will have `cc_library` as a global suggestion, when it can only be used as `native.cc_library`?

Comment (hauserx) [reply src/bazel.rs:1111]: This PR does not touch global suggestion - those are handled mostly by `get_environment` method which was not changed, just refactored a litle. The `cc_library` is suggested by `bazel-lsp` also before this PR, and this commented out code was actually passing tests before this PR.

I have changed mostly `parse_file_with_contents` which is responsible for diagnostic. To make it more clear, I have further simplified the code - renamed `go` function to `lint_module` and removed eval.rs completely (as it's not used).

**Pull request merged**